ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestConfig.java

```diff
@@ -21,26 +21,101 @@
 import io.confluent.ksql.util.KsqlServerException;
-import io.confluent.rest.RestConfig;
 import java.net.InetAddress;
 import java.net.MalformedURLException;
 import java.net.URL;
 import java.net.UnknownHostException;
+import java.util.Arrays;
+import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 import java.util.function.Function;
+import org.apache.kafka.common.config.AbstractConfig;
 import org.apache.kafka.common.config.ConfigDef;
 import org.apache.kafka.common.config.ConfigDef.Importance;
 import org.apache.kafka.common.config.ConfigDef.Type;
+import org.apache.kafka.common.config.ConfigDef.ValidString;
 import org.apache.kafka.common.config.ConfigException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class KsqlRestConfig extends RestConfig {
+public class KsqlRestConfig extends AbstractConfig {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(KsqlRestConfig.class);
 
+  public static final String LISTENERS_CONFIG = "listeners";
+  protected static final String LISTENERS_DOC =
+      "List of listeners. http and https are supported. Each listener must include the protocol, "
+          + "hostname, and port. For example: http://myhost:8080, https://0.0.0.0:8081";
+  protected static final String LISTENERS_DEFAULT = "http://0.0.0.0:8088";
+
+  public static final String AUTHENTICATION_SKIP_PATHS = "authentication.skip.paths";
+  public static final String AUTHENTICATION_SKIP_PATHS_DOC = "Comma separated list of paths that "
+      + "can be "
+      + "accessed without authentication";
+  public static final String AUTHENTICATION_SKIP_PATHS_DEFAULT = "";
+
+  public static final String ACCESS_CONTROL_ALLOW_ORIGIN_CONFIG = "access.control.allow.origin";
+  protected static final String ACCESS_CONTROL_ALLOW_ORIGIN_DOC = "Set value for "
+      + "Access-Control-Allow-Origin header";
+  protected static final String ACCESS_CONTROL_ALLOW_ORIGIN_DEFAULT = "";
+
+  public static final String ACCESS_CONTROL_ALLOW_METHODS = "access.control.allow.methods";
+  protected static final String ACCESS_CONTROL_ALLOW_METHODS_DOC = "Set value to "
+      + "Access-Control-Allow-Origin header for specified methods";
+  protected static final String ACCESS_CONTROL_ALLOW_METHODS_DEFAULT = "";
+
+  public static final String ACCESS_CONTROL_ALLOW_HEADERS = "access.control.allow.headers";
+  protected static final String ACCESS_CONTROL_ALLOW_HEADERS_DOC = "Set value to "
+      + "Access-Control-Allow-Origin header for specified headers. Leave blank to use "
+      + "default.";
+  protected static final String ACCESS_CONTROL_ALLOW_HEADERS_DEFAULT = "";
+
+  public static final String AUTHENTICATION_METHOD_CONFIG = "authentication.method";
+  public static final String AUTHENTICATION_METHOD_NONE = "NONE";
+  public static final String AUTHENTICATION_METHOD_BASIC = "BASIC";
+  public static final String AUTHENTICATION_METHOD_BEARER = "BEARER";
```

Comment:
We don't actually support this, right? I think it can be removed (and from the validator and doc string too).

Reply:
We use the config in the convert function, and convert it to the api server config equivalent. But.... in the next PR it all gets consolidated anyway.

Reply:
The next PR does not address my question. The only values for this config we support are `NONE` and `BASIC`. We should remove `BEARER` here, and also from the validator for this config, as well as the doc string.

Reply:
Ack, addressed in follow ups to next PR.